daemon/network.go

```diff
@@ -137,18 +142,22 @@ func (daemon *Daemon) enqueueIngressJob(job *ingressJob) {
 }
 
 // SetupIngress setups ingress networking.
-func (daemon *Daemon) SetupIngress(create clustertypes.NetworkCreateRequest, nodeIP string) error {
+// If caller wants to be notified when the job is completed,
+// it needs to pass a done channel.
+func (daemon *Daemon) SetupIngress(create clustertypes.NetworkCreateRequest, nodeIP string, done chan struct{}) error {
```

Comment:
How about having `SetupIngress` return a channel instead of taking one as an argument? It looks like `SetupIngress` is called rarely, and the overhead of allocating an extra channel when it's not needed would be negligible. The function signature could be:

```
func (daemon *Daemon) SetupIngress(create clustertypes.NetworkCreateRequest, nodeIP string) (<-chan struct, error) {
```